x/sync/abci.go

```diff
@@ -43,6 +45,11 @@ func EndBlocker(ctx sdk.Context, keeper Keeper) {
 				change.Status = StatusFailed
 				tagValue = types.AttributeValueChangeFailed
 			} else {
+				if change.Rewardable {
+					initiatorEthAddr := validatorsByAddr[sdk.ValAddress(change.Initiator).String()].Description.Identity
+					keeper.AddReward(ctx, initiatorEthAddr, miningReward, validator.MiningReward)
+				}
```

Comment:
could you please explain more about how the reward is distributed? e.g., how does it compare to the common block mining reward?

Reply:
The common block mining reward is distributed to all validators (including delegators) based on their stakes. The puller reward is only distributed to the puller (including delegators)

Reply:
where is the puller reward amount calculated/specified?

Reply:
https://github.com/celer-network/sgn/pull/238/files#diff-1b0d8b3cbf9287f243664d1846fa2ae4R26

Reply:
puller reward should be different from mining reward, need a separate config parameter

Reply:
I think it may not be necessary to have a separate config parameter

Reply:
it's important to have separate config as they are different rewards and may have very different weights.

Reply:
we can discuss in the meeting

Reply:
updated